retinanet/dali.py

```diff
@@ -83,50 +100,71 @@ def __repr__(self):
         ])
 
     def __len__(self):
-        return ceil(len(self.local_ids) / self.batch_size)
+        return ceil(len(self.ids) // self.world / self.batch_size)
 
     def __iter__(self):
         for _ in range(self.__len__()):
-            data, ratios, ids = [], [], []
-            dali_data, dali_ids = self.pipe.run()
+
+            data, ratios, ids, num_detections = [], [], [], []
+            dali_data, dali_boxes, dali_labels, dali_ids, dali_attrs, dali_resize_img = self.pipe.run()
+
+            for l in range(len(dali_boxes)):
+                num_detections.append(dali_boxes.at(l).shape[0])
+
+            pyt_targets = -1 * torch.ones([len(dali_boxes), max(max(num_detections),1), 5])
```

Comment:
If there was no detection for any target, is having a -1 filled detection potentially an issue?

Reply:
Won't it be ignored per https://github.com/NVIDIA/retinanet-examples/blob/master/retinanet/model.py#L138 since the label value is -1?

This is also similar as far as I see with what we do for the regular dataloader: https://github.com/NVIDIA/retinanet-examples/blob/master/retinanet/data.py#L101
https://github.com/NVIDIA/retinanet-examples/blob/master/retinanet/data.py#L111